### ISSUE TITLE
About equation (8)

### ISSUE THREAD

**Issue by A11en0** [Networks-Learning/improve-expert-predictions-conformal-prediction#2]
After reading Kerrigan et al.'s (2021) original paper, I encountered an issue with equation (8). The inference function in the model combines output logits and a confusion matrix. However, your paper only includes the confusion matrix.

Regarding the implementation, it divides by the prediction sets in the denominator. This can be seen in the following image extracted from the function error_give_test_set_per_a():

![Image](https://github.com/Networks-Learning/improve-expert-predictions-conformal-prediction/assets/33925232/2f66a27e-bfc0-4447-891f-9ac7d8b67928)

could you please help me identify the issue, thank you very much!

**Comment by ElenStr** [comment]
Please note that in our work we **do not** implement the method in Kerrigan et al.— which combines the predictions of humans alone (confusion matrix) and the predictions of the classifier in isolation (the classifier output logits). 

In our work, we use conformal prediction to compute a prediction set (by using the classifier output logits) for each data sample and then use the MNL model (a discrete choice model) to simulate the distribution of the human predictions *from the prediction set*. Eq. 8 is the formula given by the MNL model to compute the probability that the human predicts the true label *from a given prediction set*. This formula requires access only to the confusion matrix of the human predictions alone and the labels included in the prediction set. Note that in this formula each term of the sum in the denominator corresponds to each label in the prediction set.

The function error_given_test_set_per_a() first simulates (via the MNL model) the categorical distribution of the human predictions from the prediction set for each data sample (line 238) and then samples the human prediction from this distribution (line 241). To simulate this distribution using the MNL model, one basically reweights the confusion matrix of the human predictions alone and computes a "new confusion matrix" for the given data sample assuming that the person can select only from the labels the prediction set.

To give a more concrete example, assume a prediction set with labels $\mathcal{C} = \\{A, B, \Gamma\\}$ for a data sample with true label $B$ and let $C_{ij}$ be the entry of the confusion matrix of the human predictions alone for the label $i$ assuming that the true label is $j$. Then the categorical distribution of the human predictions from the prediction set would be (what line 238 would compute) $P[\hat{Y} = A ; \mathcal{C} | A \in \mathcal{C}] = C_{AB} / (C_{AB} + C_{BB} + C_{\Gamma B})$,  $P[\hat{Y} = B ; \mathcal{C} | B \in \mathcal{C}] = C_{BB} / (C_{AB} + C_{BB} + C_{\Gamma B})$ and  $P[\hat{Y} = \Gamma ; \mathcal{C} | \Gamma \in \mathcal{C}] = C_{\Gamma B} / (C_{AB} + C_{BB} + C_{\Gamma B})$. The variable ```denominators``` in the code would compute the denominators of the previous three probabilities, and the highlighted variable ```sets``` is a 0-1 matrix, with 1 for the labels $A, B, \Gamma$ included in the prediction set and 0 for all other labels. The variable ```sets``` is essentially a mask used to select the terms $C_{AB}, C_{BB}$ and $C_{\Gamma B}$.

Happy to help if there are more questions.
